### PR TITLE
Add missing members to Path

### PR DESCRIPTION
**Pull request by stephentoub**
Path is missing the Combine(string, string, string, string) overload and the obsolete InvalidPathChars field.

cc: @JeremyKuhne, @danmosemsft, @ianhays

### CODE REVIEW

File: src/System.Runtime.Extensions/src/System/IO/Path.cs [hunk 1]
```diff
@@ -214,6 +217,20 @@ public static string Combine(string path1, string path2, string path3)
             return CombineNoChecks(path1, path2, path3);
         }
 
+        public static string Combine(string path1, string path2, string path3, string path4)
+        {
+            if (path1 == null || path2 == null || path3 == null || path4 == null)
+                throw new ArgumentNullException((path1 == null) ? "path1" : (path2 == null) ? "path2" : (path3 == null) ? "path3" : "path4");
+            Contract.EndContractBlock();
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/src/System/IO/Path.cs]: >Contract.EndContractBlock(); [](start = 12, length = 28)

are we still using Contract? or just to be consistent with the other methods here?

**Comment by danmosemsft** [comment]
Thank you for doing this!

**Comment by danmosemsft** [comment]
:shipit:

File: src/System.Runtime.Extensions/src/System/IO/Path.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime.Extensions/src/System/IO/Path.cs]: I did it just for consistency with the rest of the implementations.  If we're going to remove them, we should do it all up I think.

**Comment by ianhays** [comment]
:shipit:

**Pull request merged**